tool/tbot/copy_binaries.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package main
+
+import (
+	"context"
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/gravitational/teleport/lib/tbot/cli"
+	"github.com/gravitational/trace"
+)
+
+const (
+	// fdpassBinaryName is the name of the fdpass binary
+	fdpassBinaryName = "fdpass-teleport"
+
+	// tbotBinaryName is the name of the tbot binary
+	tbotBinaryName = "tbot"
+)
+
+// copyBinary copies a binary from the source to the destination. It assumes
+// 0755 permissions.
+func copyBinary(src, dest string) error {
+	inputFile, err := os.Open(src)
+	if err != nil {
+		return trace.Wrap(err, "opening source file: %s", src)
+	}
+	defer inputFile.Close()
+
+	destFile, err := os.OpenFile(dest, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0755)
+	if err != nil {
+		return trace.Wrap(err, "opening destination for writing: %s", dest)
+	}
+	defer destFile.Close()
+
+	_, err = io.Copy(destFile, inputFile)
+	if err != nil {
+		return trace.Wrap(err, "copying file contents: %s", src)
+	}
+
+	return nil
+}
+
+// onCopyBinariesCommand runs `tbot copy-binaries`
+func onCopyBinariesCommand(ctx context.Context, cmd *cli.CopyBinariesCommand) error {
+	selfPath, err := os.Executable()
```

Comment:
The only thing we should double check is if this will run into any weirdness with auto-upgrades

Reply:
It's definitely susceptible but at least we don't do any re-exec shenanigans in tbot so it should basically always be "fine", and will be a nonissue for Docker, GHA, etc where we don't use auto upgrades at all.

My assumption is that we'll only copy the wrong binary if it gets upgraded out from under us (which may never happen and may not be allowed depending on OS/filesystem/etc). Since `tbot` never reexecs to match cluster versions it at least reliably executes as whatever is pointed to via $0, and will copy whatever exists at that same path. Not 100% foolproof but I don't think we can implement this much better without depending on Linux specific `/proc/self` semantics.